src/main.cpp

```diff
@@ -1,89 +1,129 @@
+        log(INFO, "allLoadedBeatmapLevelPackCollection pointer: %p", allLoadedBeatmapLevelPackCollection);
+        // Since we know that allLoadedBeatmapLevelPackCollection is actually of type: IBeatmapLevelPackCollection, we can use it like one
+        auto* beatmapLevelPacks = CRASH_UNLESS(il2cpp_utils::GetPropertyValue<Il2CppArray*>(
+            allLoadedBeatmapLevelPackCollection, "beatmapLevelPacks"));
+        log(INFO, "There are a total of: %u level packs", il2cpp_functions::array_length(beatmapLevelPacks));
+    }
+    return actualIsLoadedValue;  // be sure to actually return the original function's result if you want the caller to have it!
 }
 
-// Hook an instance property getter: BeatmapLevelsModel.get_customLevelPackCollection()
+// Generally /avoid/ hooking instance property getters like BeatmapLevelsModel.get_customLevelPackCollection()
+// A hook needs 3-5 instructions, so hooking methods with too few instructions can result in the next method being overwritten!
 MAKE_HOOK_OFFSETLESS(BeatmapLevelsModel_get_customLevelPackCollection, Il2CppObject*, Il2CppObject* self) {
-    // We can call other methods whenever we want:
-    Il2CppObject* allLoadedBeatmapLevelPackCollection;
-    if (!il2cpp_utils::RunMethod(&allLoadedBeatmapLevelPackCollection, self, "get_allLoadedBeatmapLevelPackCollection")) {
-        // Failed to run method!
-        log(CRITICAL, "Failed to run method: get_allLoadedBeatmapLevelPackCollection on BeatmapLevelsModel!");
-    }
-    log(INFO, "allLoadedBeatmapLevelPackCollection pointer: %p", allLoadedBeatmapLevelPackCollection);
-    // Since we know that allLoadedBeatmapLevelPackCollection is actually of type: IBeatmapLevelPackCollection, we can use it like one
-    Il2CppArray* beatmapLevelPacks;
-    if (!il2cpp_utils::RunMethod(&beatmapLevelPacks, allLoadedBeatmapLevelPackCollection, "get_beatmapLevelPacks")) {
-        // Failed to run method!
-        log(CRITICAL, "Failed to run method: get_beatmapLevelPacks of allLoadedBeatmapLevelPackCollection!");
-    }
-    log(INFO, "There are a total of: %ui level packs", il2cpp_functions::array_length(beatmapLevelPacks));
     return BeatmapLevelsModel_get_customLevelPackCollection(self);
 }
 
 // Hook a void static method: MonoBehaviour.print(object message)
 MAKE_HOOK_OFFSETLESS(MonoBehaviour_print, void, Il2CppObject* message) {
     // We can access other static things too, just as usual.
     // For now, however, we will choose to simply perform nothing.
-    // We do not have to call the original function if we do not want to, although it may cause code to crash in strange ways
+    // We do not have to call the original function if we do not want to, although it may cause code to crash in strange ways.
     return;
 }
 
-// This function is called when the mod is loaded for the first time
-__attribute__((constructor)) void lib_main()
-{
+// This attribute means the function will be called when the mod is loaded for the first time.
```

Comment:
Attribute, yes. The function is _also_ `dlsym`'d, so that might be worth remembering